api/src/main/java/io/strimzi/api/kafka/model/nodepool/KafkaNodePoolSpec.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka.model.nodepool;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import io.fabric8.kubernetes.api.model.ResourceRequirements;
+import io.strimzi.api.kafka.model.Constants;
+import io.strimzi.api.kafka.model.JvmOptions;
+import io.strimzi.api.kafka.model.Spec;
+import io.strimzi.api.kafka.model.storage.Storage;
+import io.strimzi.crdgenerator.annotations.Description;
+import io.strimzi.crdgenerator.annotations.KubeLink;
+import io.strimzi.crdgenerator.annotations.Minimum;
+import io.sundr.builder.annotations.Buildable;
+import lombok.EqualsAndHashCode;
+
+import java.util.List;
+
+/**
+ * Represents the .spec section of a KafkaNodePool resource
+ */
+@Buildable(
+        editableEnabled = false,
+        builderPackage = Constants.FABRIC8_KUBERNETES_API
+)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"replicas", "storage", "roles", "resources", "jvmOptions", "template"})
+@EqualsAndHashCode(callSuper = true)
+public class KafkaNodePoolSpec extends Spec {
+    private static final long serialVersionUID = 1L;
+
+    private int replicas;
+    private Storage storage;
+    private List<ProcessRoles> roles;
+    private ResourceRequirements resources;
+    private JvmOptions jvmOptions;
+    private KafkaNodePoolTemplate template;
+
+    @Description("The number of pods in the pool.")
+    @Minimum(0)
```

Comment:
I was wondering if we should allow to have an empty node pool. What does it mean? The current minimum `spec.replicas` we allow on Kafka and Zookeeper is 1. How does the node pool differ from this perspective?

Reply:
The main difference is that you can have multiple node pools. So in theory, you can have one with 0 replicas (=> no Kafka nodes) and another one with >0 replicas. Thus the cluster would still have some nodes and can work. Where is if you set replicas to 0 in the Kafka CR, it would mean 0 brokers and that would not really work.

Reply:
Yeah but I was just wondering about what's the use case of an empty node pool. Thinking more maybe it allows the user to move brokers from a pool to another, making the first empty and then deleting it if needed. Or maybe they can take a node pool definition because will add more brokers later. With node pools (even if empty) they can have a kind of high-level vision of where brokers can be deployed and what are the resources that each pool has (storage, CPU, ..).